weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx

```diff
@@ -47,68 +75,238 @@ export const CallsPage: FC<{
   paginationModel: GridPaginationModel;
   setPaginationModel: (newModel: GridPaginationModel) => void;
 }> = props => {
+  const {entity, project, baseView, views, fetchViews} = props;
   const [filter, setFilter] = useControllableState(
     props.initialFilter ?? {},
     props.onFilterUpdate
   );
 
   const isEvaluationTable = useCurrentFilterIsEvaluationsFilter(
     filter,
-    props.entity,
-    props.project
+    entity,
+    project
   );
+  // table is the internal id stored in the object.
+  const table = isEvaluationTable ? 'evaluations' : 'traces';
+
+  const getTsClient = useGetTraceServerClientContext();
+
+  const tsClient = getTsClient();
+  const view = baseView.object_id;
 
-  const title = useMemo(() => {
-    if (isEvaluationTable) {
-      return 'Evaluations';
+  const currentViewDefinition = useCurrentViewDefinition(baseView);
+
+  const history = useHistory();
+  const onLoadView = (viewToLoad: TraceObjSchema) => {
+    // We want to preserve any params that are not part of view definition,
+    // e.g. peek drawer state.
+    const newQuery = new URLSearchParams(history.location.search);
+
+    // Clear out any params related to saved views
+    for (const key of SAVED_PARAM_KEYS) {
+      newQuery.delete(key);
     }
-    if (filter.opVersionRefs?.length === 1) {
-      const opName = opVersionRefOpName(filter.opVersionRefs[0]);
-      if (opName) {
-        return opNiceName(opName) + ' Traces';
-      }
+
+    // Update with params from the view definition
+    for (const [key, value] of Object.entries(viewToLoad.val.definition)) {
+      newQuery.set(key, JSON.stringify(value));
+    }
+
+    newQuery.set('view', viewToLoad.object_id);
+    history.push({search: newQuery.toString()});
+    props.onRecordLastView(viewToLoad.object_id);
+  };
+  const onResetView = () => {
+    let viewToLoad = views?.find(v => v.object_id === view);
+    if (!viewToLoad) {
+      const defaultViewId = getDefaultViewId(table);
+      viewToLoad = views?.find(v => v.object_id === defaultViewId);
+    }
+    if (viewToLoad) {
+      onLoadView(viewToLoad);
+    } else {
+      console.error('could not find view to reset to');
     }
-    return 'Traces';
-  }, [filter.opVersionRefs, isEvaluationTable]);
+  };
+
+  const {onRecordLastView} = props;
+
+  const onUpsertView = useCallback(
+    (objectId: string, label: string | null, successMessage: string) => {
+      if (objectId === getDefaultViewId(table)) {
+        objectId = getNewViewId(table);
+      }
+      if (label === null) {
+        // If caller doesn't provide a new label, use the existing one.
+        label = baseView.val.label;
+      }
+      const className = 'SavedView';
+      tsClient
+        .objCreate({
```

Comment:
As implemented, because `SavedView` is not defined on the server, this has no validation. This makes me a bit nervous that this critical data structure is enforced within this component. If you defined a builtin object, you could use `useCreateBuiltinObjectInstance` which would allow you to simply pass the relevant data:
```
    project_id: string;
    object_id: string;
    val: T;
    builtin_object_class?: string;
```

And have val be validated in typescript and enforced on the backend. This feels like a safer approach